modules/v2x/v2x_proxy/app/v2x_proxy.cc

```diff
@@ -13,147 +13,211 @@
+    while (!exit_) {
+      this->recv_trafficlight();
+    }
+  }));
+  CHECK(!!_hdmap && !!v2x_carstatus_timer_ && !!os_interface_ &&
+        !!obu_interface_ && os_interface_->InitFlag() &&
+        obu_interface_->InitFlag());
+  // FIXME: NEED REWRITE PROGRAM TO FIX IT!
+  planning_thread_.reset(new std::thread([this]() {
+    while (!exit_) {
+      this->recv_os_planning();
+    }
+  }));
   v2x_carstatus_timer_->Start();
-
+  //    ::grpc::ServerBuilder builder;
+  //    builder.AddListeningPort(
+  //        FLAGS_grpc_server_host + ":" + FLAGS_grpc_debug_server_port,
+  //        ::grpc::InsecureServerCredentials());
+  //    builder.RegisterService(_m_service_debug.get());
+  //    _m_server = builder.BuildAndStart();
+  /* Monitor timer to obu status */
+  AINFO << "_md5 " << _md5;
+  get_rsulist_from_file(FLAGS_rsu_whitelist_name, &rsu_list_);
   init_flag_ = true;
 }
 
-bool V2xProxy::InitFlag() { return init_flag_; }
-
-bool V2xProxy::TrafficLightProc(CurrentLaneTrafficLight* msg) {
-  if (!msg->has_gps_x_m() || !msg->has_gps_y_m()) {
-    AERROR << "Error::v2x trafficlight ignore, gps point is null";
-    return false;
+bool V2xProxy::get_rsulist_from_file(const std::string &filename,
+                                     std::set<std::string> *whitelist) {
+  RETURN_VAL_IF_NULL(whitelist, false);
+  std::ifstream input_file(filename);
+  RETURN_VAL_IF(!input_file, false);
+  bool flag = false;
+  std::string strOne;
+  while (getline(input_file, strOne)) {
+    flag = true;
+    whitelist->insert(strOne);
   }
-  apollo::common::PointENU point;
-  point.set_x(msg->gps_x_m());
-  point.set_y(msg->gps_y_m());
-  std::vector<apollo::hdmap::SignalInfoConstPtr> signals;
-  if (hdmap_->GetForwardNearestSignalsOnLane(point, 1000.0, &signals) != 0) {
-    AERROR << "Error::v2x trafficlight ignore, hdmap get no signals";
-    AERROR << "traffic light size : " << signals.size();
-    return false;
-  }
-  for (auto i = signals.begin(); i != signals.end(); i++) {
-    if ((*i)->id().id().empty()) {
-      AERROR << "Error::v2x trafficlight ignore, signals id is empty";
-      return false;
+  return flag;
+}
+
+bool V2xProxy::init_flag() { return init_flag_; }
+
+void V2xProxy::recv_os_planning() {
+  auto planning_adc = std::make_shared<::apollo::planning::ADCTrajectory>();
+  auto res_light =
+      std::make_shared<::apollo::perception::TrafficLightDetection>();
+  os_interface_->get_planning_adc_from_os(planning_adc);
+  // OK GET PLANNING INFO
+  std::shared_ptr<::v2x::proxy::OSLightT> last_os_light = nullptr;
+  {
+    std::lock_guard<std::mutex> lock(lock_last_os_light_);
+    struct timespec ts = {};
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+    if (last_os_light_ == nullptr ||
+        2000LL * 1000 * 1000 <
+            ::v2x::proxy::utils_diff_timespec(&ts, &ts_last_os_light_)) {
+      AWARN << "V2X Traffic Light is too old!";
+      last_os_light_ = nullptr;
+    } else {
+      AINFO << "V2X Traffic Light is on time.";
```

Comment:
Avoid using AINFO to minimize log flooding. AINFO can only be used in initialization process for checking purposes.